airflow/models/xcom_arg.py

```diff
@@ -309,11 +310,26 @@ def zip(self, *others: XComArg, fillvalue: Any = NOTSET) -> ZipXComArg:
         return super().zip(*others, fillvalue=fillvalue)
 
     def get_task_map_length(self, run_id: str, *, session: Session) -> int | None:
+        from airflow.models.taskinstance import TaskInstance
         from airflow.models.taskmap import TaskMap
         from airflow.models.xcom import XCom
 
         task = self.operator
         if isinstance(task, MappedOperator):
+            unfinished_ti_count_query = session.query(func.count(TaskInstance.map_index)).filter(
```

Comment:
Do you really need the count here? I think .first() is enough and it can help to avoid scanning all the mapped tis at each query. WDYT?

Reply:
TI is indexed in this query so scanning should not happen, and having `COUNT` reduces data sent through the wire. IIRC we did a micro benchmark against a similar query sometime back; the best choice in theory is `EXISTS` but it’s not universally supported. The second best is `SELECT 1 FROM ...` + `first()` but that reads pretty weird, and `COUNT` is the next best choice.

Reply:
Maybe we should add a helper that encapsulate this logic (or use `EXISTS` when available). There are actually quite many use cases.